src/System.IO.Pipelines.Networking.Tls/Internal/OpenSsl/CustomBio.cs

```diff
@@ -0,0 +1,256 @@
+            s_methodPtr = Marshal.AllocHGlobal(sizeToAlloc);
+            Marshal.StructureToPtr(s_methodStruct, s_methodPtr, false);
+        }
+
+        public static IntPtr Custom() => s_methodPtr;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int Create(ref bio_st bio);
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int Write(ref bio_st bio, void* buf, int num);
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int Read(ref bio_st bio, void* buf, int size);
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate long Control(ref bio_st bio, BioControl cmd, long num, void* ptr);
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int Free(ref bio_st data);
+
+        private static class WindowsLib
+        {
+            public const string CryptoDll = "libeay32.dll";
+            [DllImport(CryptoDll, CallingConvention = CallingConvention.Cdecl)]
+            public extern static void BIO_set_flags(ref bio_st bio, BioFlags flags);
+        }
+
+        private static class UnixLib
+        {
+            public const string CryptoDll = "libcrypto.so";
+            [DllImport(CryptoDll, CallingConvention = CallingConvention.Cdecl)]
+            public extern static void BIO_set_flags(ref bio_st bio, BioFlags flags);
+        }
+
+        private static class OsxLib
+        {
+            public const string CryptoDll = "libcrypto.1.0.0.dylib";
+            [DllImport(CryptoDll, CallingConvention = CallingConvention.Cdecl)]
+            public extern static void BIO_set_flags(ref bio_st bio, BioFlags flags);
+        }
+
+        private static void BIO_set_flags(ref bio_st bio, BioFlags flags)
+        {
+            if (Interop.IsWindows)
+            {
+                WindowsLib.BIO_set_flags(ref bio, flags);
+            }
+            else if (Interop.IsOsx)
+            {
+                OsxLib.BIO_set_flags(ref bio, flags);
+            }
+            else
+            {
+                UnixLib.BIO_set_flags(ref bio, flags);
+            }
+        }
+
+        private static int FreeBio(ref bio_st bio)
+        {
+            if (bio.init == 1 && bio.shutdown == 1)
+            {
+                bio.ptr = null;
+                return 1;
+            }
+            return -1;
+        }
+
+        private static int CreateBio(ref bio_st bio)
+        {
+            bio.shutdown = 1;
+            bio.init = 1;
+            bio.num = -1;
+            return 1;
+        }
+
+        private static int WriteBio(ref bio_st bio, void* buff, int numberOfBytes)
+        {
+            var bufferHandle = GCHandle.FromIntPtr((IntPtr)bio.ptr);
+            var bufferPipeline = (IPipelineWriter)bufferHandle.Target;
+            int numberOfBytesRemaing = numberOfBytes;
+            var buffer = bufferPipeline.Alloc();
```

Comment:
Agreed, as explained this was changed because the ASP.NET team wanted it, when it was in that domain. I am in line with var only where it is clear what the "var" will be